gdbgui: init at 0.9.0.1

yrashk opened the pull request:
Feature rich browser-based frontend with data structure visualizations
(like DDD), and gdb terminal access. Compatible with C, C++, golang,
Rust, fortran. Written in Python and JavaScript

gdbgui is at the top of the GDB Front Ends
list on https://sourceware.org/gdb/wiki/GDB%20Front%20Ends

###### Motivation for this change

gdbgui is not present in nixpkgs

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

yrashk commented:
This is my first "new package" PR so please let me know if I missed some requirement or convention. Thank you!

bjornfor commented:
Cool!

It doesn't build for me (test phase fails):

```
$ nix-build -A gdbgui
...
wrapping `/nix/store/didsnhp5ixi1a9gqg6a588zv0gpqa0jr-gdbgui-0.9.0.1/bin/.gdbgui-wrapped'...
running install tests
running test
error: protocol not found
builder for ‘/nix/store/0rh11mh4lk772xs67k7x0y8f2xfwgqs2-gdbgui-0.9.0.1.drv’ failed with exit code 1
error: build of ‘/nix/store/0rh11mh4lk772xs67k7x0y8f2xfwgqs2-gdbgui-0.9.0.1.drv’ failed
```

Can you try with "nix.useSandbox = true" and see if you get the same issue? Looks like the tests does something that is unavailable in Nix sandboxes.

Mic92 commented:
Looking at the error message I assume it tries to access `/etc/protocols` or `/etc/services`.
There is very ugly workaround for that: https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/python-packages.nix#L2386
Maybe the test can be patched to disable that or if it is a single test it can be skipped.

yrashk committed:
gdbgui: init at 0.9.0.1

Feature rich browser-based frontend with data structure visualizations
(like DDD), and gdb terminal access. Compatible with C, C++, golang,
Rust, fortran. Written in Python and JavaScript

gdbgui is at the top of the GDB Front Ends
list on https://sourceware.org/gdb/wiki/GDB%20Front%20Ends

Mic92 commented on pkgs/development/tools/misc/gdbgui/default.nix:
```diff
@@ -0,0 +1,40 @@
+    preCheck = ''
+      export NIX_REDIRECTS=/etc/protocols=${pkgs.iana-etc}/etc/protocols \
+             LD_PRELOAD=${pkgs.libredirect}/lib/libredirect.so
+    '';
```
This fixes tests for me as well. The only disadvantage I see that this method is not available on darwin (and maybe not required?). Can you make this optional on linux:

```
preCheck = lib.optionalString stdenv.isLinux ''
      export NIX_REDIRECTS=/etc/protocols=${pkgs.iana-etc}/etc/protocols \
             LD_PRELOAD=${pkgs.libredirect}/lib/libredirect.so
'';
```

then we can check with our pr bot, if it builds on darwin.

yrashk replied:
Thanks, amended!

Mic92 commented:
@GrahamcOfBorg build gdbgui

GrahamcOfBorg reviewed (commented):
Success for system: x86_64-darwin

```
patching script interpreter paths in /nix/store/f1q2kd9apxbwwhdc0fnmddldnhf2mfxh-gdbgui-0.9.0.1
wrapping `/nix/store/f1q2kd9apxbwwhdc0fnmddldnhf2mfxh-gdbgui-0.9.0.1/bin/.gdbgui-wrapped'...
running install tests
running test
..
----------------------------------------------------------------------
Ran 2 tests in 0.024s

OK
/nix/store/f1q2kd9apxbwwhdc0fnmddldnhf2mfxh-gdbgui-0.9.0.1
```

Mic92 commented:
Ok green light on darwin. Thanks

GrahamcOfBorg reviewed (commented):
Success for system: x86_64-linux

```
checking for references to /tmp/nix-build-gdbgui-0.9.0.1.drv-0 in /nix/store/c6bidhm170discbv72j77rrshyj4kh5g-gdbgui-0.9.0.1...
wrapping `/nix/store/c6bidhm170discbv72j77rrshyj4kh5g-gdbgui-0.9.0.1/bin/.gdbgui-wrapped'...
running install tests
running test
..
----------------------------------------------------------------------
Ran 2 tests in 0.153s

OK
/nix/store/c6bidhm170discbv72j77rrshyj4kh5g-gdbgui-0.9.0.1
```
